### PR TITLE
Hl 1093 y2k24 bug

### PR DESCRIPTION
**Pull request by rikuke**
## Description :sparkles:
Fix application start date validations that did not allow the start date to be in the past year.
Fix validations to allow a start date 4 months in the past.

### CODE REVIEW

File: frontend/benefit/applicant/src/components/applications/forms/application/step2/utils/validation.ts [hunk 1]
```diff
@@ -45,9 +46,9 @@ export const getValidationSchema = (
       .required(t(VALIDATION_MESSAGE_KEYS.REQUIRED))
       .test({
         message: t(VALIDATION_MESSAGE_KEYS.DATE_MIN, {
-          min: convertToUIDateFormat(startOfYear(new Date())),
+          min: convertToUIDateFormat(subMonths(new Date(), 4)),
         }),
-        test: (value = '') => validateDateIsFromCurrentYearOnwards(value),
+        test: (value = '') => validateIsTodayOrPastDate(value),
```
Comment (sirtawast) [inline frontend/benefit/applicant/src/components/applications/forms/application/step2/utils/validation.ts:51]: 👍

**Review by sirtawast — APPROVED** [review]

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2684.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2684.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2684.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2684.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2684.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**